pkg/router/template/plugin.go

```diff
@@ -265,9 +265,27 @@ func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, lookupSvc
 		wasIdled = true
 	}
 
-	out := make([]Endpoint, 0, len(endpoints.Subsets)*4)
+	// Count the number of endpoint addresses (we have to do this first since we need the result when we loop below)
+	// But the two loop conditions must be the same
+	numEndpoints := 0
+	for _, s := range subsets {
```

Comment:
Run this loop only if wasIdled=false. Just to be more efficient.

Reply:
Or rather run a loop in the end on the 'out' array only if wasIdled=false and len(out)==1 and fix all elements of the 'out' array to set NoHealthCheck=true. This way we would only loop if the length is 1.

Reply:
Fair enough... but I'm also considering adding metrics to the router for:
 - Num idled
 - Num endpoints
 - Num health checked endpoints

And so on... so may need to flip this back.  However, until we do that, I'll go for the faster version.